runtime/autoload/health.vim

```diff
@@ -1,433 +1,181 @@
-          \ . 'using Zsh, see: http://vi.stackexchange.com/a/7654/5229')
+""
+" s:add_single_checker is a function to handle adding a checker of name
+" {checker_name} to the list of health_checkers. It also enables it.
+function! s:add_single_checker(checker_name) abort
+  if has_key(g:health_checkers, a:checker_name)
+    " TODO: What to do if it's already there?
+    return
+  else
+      let g:health_checkers[a:checker_name] = v:true
   endif
+endfunction
 
-  " Diagnostic output
-  echo 'Checking: Python' a:version
-  echo '  Executable:' (empty(python_bin) ? 'Not found' : python_bin)
-  if len(python_multiple)
-    for path_bin in python_multiple
-      echo '     (other):' path_bin
+""
+" s:add_checker is a function to register at least one healthcheckers.
+" {checker_name} can be specified by either a list of strings or a single string.
+" The string should be the name of the function to check, which should follow
+" the naming convention of `health#plugin_name#check`
+function! s:add_checker(checker_name) abort " {{{
+  if type(a:checker_name) == type('')
+    call s:add_single_checker(a:checker_name)
+  elseif type(a:checker_name) == type([])
+    for checker in a:checker_name
+      call s:add_single_checker(checker)
     endfor
   endif
+endfunction " }}}
 
-  if !empty(python_bin)
-    let [pyversion, current, latest, status] = s:version_info(python_bin)
-    if a:version != str2nr(pyversion)
-      call add(notes, 'Warning: Got an unexpected version of Python.  '
-            \ . 'This could lead to confusing error messages.  Please '
-            \ . 'consider this before reporting bugs to plugin developers.')
-    endif
-    if a:version == 3 && str2float(pyversion) < 3.3
-      call add(notes, 'Warning: Python 3.3+ is recommended.')
-    endif
-
-    echo '  Python Version:' pyversion
-    echo printf('  %s-neovim Version: %s', python_bin_name, current)
-
-    if current == 'not found'
-      call add(notes, 'Error: Neovim Python client is not installed.')
-    endif
-
-    if latest == 'unknown'
-      call add(notes, 'Warning: Unable to fetch latest Neovim Python client version.')
-    endif
-
-    if status == 'outdated'
-      echon ' (latest: '.latest.')'
-    else
-      echon ' ('.status.')'
-    endif
+function! health#disable_checker(checker_name) abort " {{{
+  if has_key(g:health_checkers, a:checker_name)
+    let g:health_checkers[a:checker_name] = v:false
+  else
+    " TODO: What to do if it's not already there?
+    return
   endif
+endfunction " }}}
 
-  call s:echo_notes(notes)
-endfunction
+function! s:change_file_name_to_health_checker(name) abort " {{{
+  return substitute(substitute(substitute(a:name, ".*autoload/", "", ""), "\\.vim", "#check", ""), "/", "#", "g")
+endfunction " }}}
 
+function! health#_default_checkers() abort " {{{
+  " Get all of the files that are in autoload/health/ folders with a vim
+  " suffix
+  let checker_files = globpath(&runtimepath, 'autoload/health/*.vim', 0, 1)
```

Comment:
We definitely want the `{nosuf}` parameter to be `1`, not `0`--otherwise `'wildignore'` et. al. applies to the search, which makes it dependent on the user's configuration.